k8s/oidc-vault/README.md

```diff
@@ -0,0 +1,435 @@
+   Threshold          3
+   Unseal Progress    1/3 # <- this is how many keys you have entered
+   Unseal Nonce       e1bf3fa2-0058-5703-e2dc-a5c45c1b7f9a
+   Version            1.3.4
+   HA Enabled         false
+   ```
+   Note that here we see a key `Sealed` that tells us that the vault has not been unsealed yet, and a key `Unseal Progress` that says how many correct Unseal Keys we have entered
+   
+   Once we entered three different correct keys we have successfully unsealed the vault and the key `Seal` must be `false`
+   
+   ```console
+   Sealed          false
+   ```
+
+### Enable secrets engine and store a test secret
+
+Using our root access via the CLI (the Initial Root Key in `VAULT_TOKEN`) we are going to enable the `kv` (key-value) secrets engine and store a secret that we are going to retrieve later using our SPIRE-enabled login.
+
+1. Given that we are probably in a different terminal window, let's set the `VAULT_ADDR` again, and the `VAULT_TOKEN` with the Initial Root Key.
+
+   ```console
+   $ export VAULT_ADDR=http://127.0.0.1:8200
+   $ export VAULT_TOKEN="s.PFuCtYgzjh6mRAfAVjfsGv3O"
+   ```
+
+2. Enable the `kv` (key-value) secrets engine on the `secret/` path
+
+   ```console
+   $ vault secrets enable -path=secret kv
+   ```
+
+3. Put a secret in the new path, this is what we are going to retrieve using our SPIRE-enabled identity.
+
+   ```console
+   $ vault kv put secret/my-super-secret test=123
+   ```
+
+### Set up Vault OIDC federation with SPIRE
+
+In this step we'll configure the Vault server to federate with our SPIRE Server running on a Kubernetes cluster.
+
+1. Enable the JWT authentication method
+
+   ```console
+   $ vault auth enable jwt
+   ```
+
+2. Set up our OIDC Discovery URL
+
+   ```console
+   $ vault write auth/jwt/config oidc_discovery_url=https://spire-vault-oidc.skitalee.org default_role=“dev”
+   ```
+
+3. Define a policy `my-dev-policy` that will be assigned to a `dev` role that we'll create in the next step.
+
+   Create a policy file called `vault-policy.hcl` with the following content (you can find the file [here](./vault/vault-policy.hcl)):
+
+   ```console
+   path "secret/my-super-secret" {
+      capabilities = ["read"]
+   }
+   ```
+
+   and create the policy
+
+   ```console
+   $ vault policy write my-dev-policy vault-policy.hcl
+   ```
+
+3. Create a role `dev`, binding the subject and audience that will be in the JWT, and configuring the `sub` claim that will be used to identify the user. Also set up a 24 hours TTL for testing purposes and a policy `my-dev-policy` that will be assigned to the tokens.
+
+   ```console
+   $ vault write auth/jwt/role/dev role_type=jwt user_claim=sub bound_audiences=TESTING bound_subject=spiffe://example.org/ns/default/sa/default token_ttl=24h token_policies=my-dev-policy
+   ```
+
+## Get Vault credentials
+
+Now we are going to get an access token to use with Vault, through SPIRE Federation using our client to get an identity.
```

Comment:
This might be simpler for beginning users if we split this into two sentences.

```suggestion
Now we are going to get an access token to use with Vault. We'll use a sample client workload to get an identity using the SPIRE Federation feature.
```